src/main/java/cz/cvut/kbss/study/service/repository/RepositoryUserService.java

```diff
@@ -243,6 +235,71 @@ protected void preUpdate(User instance) {
                         currentUser.getUsername()));
             }
         }
+    }
+
+    protected void validateRecordsAgainstCollisions(User toUpdate, User original) {
```

Comment:
@palagdan, look at my last commit. I had the following issues:
- 1) to me, variable names were not correct, e.g,. newInsitutionRecords ==> existingInstitutionRecords
- 2) although not that important -- algorithm was of higher complexity than needed because of comparing every label with every other label, i.e., on average O(N^2)
- 3) IMPORTANT: I wanted to have a more detailed message on errors (although we don't have parametrized messages i would like to get there someday :)

Reply:
- I think there might be problem within UI how to show this message due to `\n` in the message and how it is interpreted in HTML (did not test it)